[export] Add setattr for ep.example_inputs

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140991
* __->__ #140990

Differential Revision: [D66136725](https://our.internmc.facebook.com/intern/diff/D66136725)

angelayi committed:
[export] Add setattr for ep.example_inputs

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66136725).

yushangdi commented on test/export/test_export.py:
```diff
@@ -3537,6 +3537,48 @@ def forward(self, arg1, arg2, a=1, b=2):
+        with self.assertRaisesRegex(ValueError, "Example inputs should be a tuple"):
+            ep.example_inputs = (torch.ones(3),)
+
+        with self.assertRaisesRegex(ValueError, "Ran into a kwarg keyword mismatch"):
```
does it work when the keyword has default values? e.g. `def forward(self, a, *, x, y, z=torch.ones(2))`. Now am I still required to provide a value for `z`?

angelayi replied:
Lol just found out that dynamo does not support default tensor kwargs:
> torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
But for non-strict, we will not treat the kwarg with a default value as an input, so you *cant* provide a value for z.

yushangdi replied:
> Lol just found out that dynamo does not support default tensor kwargs:
> 
> > torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
> > But for non-strict, we will not treat the kwarg with a default value as an input, so you _cant_ provide a value for z.

lol I see.

angelayi committed:
Update on "[export] Add setattr for ep.example_inputs"


Differential Revision: [D66136725](https://our.internmc.facebook.com/intern/diff/D66136725)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66136725).
